bigchaindb_driver/transport.py

```diff
@@ -74,14 +74,23 @@ def forward_request(self, method, path=None,
                     headers=headers
                 )
                 return response.data
+            except TransportError as err:
+                end = time()
+                time_left -= timedelta(seconds=end - start)
+                self.pool.fail_node()
+                start = time()
+                code = None if err.status_code is None else err.status_code
+                info = None if err.info is None else err.info
+                exceptions[err.url] = {"code": code, "info": info}
```

Comment:
I would collect them into a list - currently we can lose some information if the same node experiences 2 different exceptions. What do you think about a list of pairs `(url, error)`?

Reply:
if we do list, we are going to repeat errors, as we would easily have the same error multiple times and i dont think we will lose relevant info, we are giving the last error why every node failed.

Reply:
Repeated errors just reflect the real history of requests and might provide useful hints. Repeating is not always bad.

The current implementation should also work just fine - you can keep it.

Reply:
Ok, updated with pair instead of dict, and keep the dict. Other suggestions added on next push.